packages/network-controller/src/NetworkController.ts

```diff
@@ -203,6 +203,11 @@ export type NetworkConfiguration = {
    * interact with the chain.
    */
   rpcEndpoints: RpcEndpoint[];
+  /**
+   * Profile Sync - Network Sync field.
+   * Allows comparison of local network state with state to sync.
+   */
+  lastUpdatedAt?: number;
```

Comment:
Do we want to deal with the fact that this could be `undefined` in the clients or in other controllers? If not, should we remove the `?` here?

I guess the tradeoffs would be:

- We will have to make sure to update `getDefaultNetworkConfigurationsByChainId`.
- We will have to write a migration in the clients to fill this value in.

Reply:
You've hit the nail on the head, we wanted a way to leverage the fact that existing users networks might not have this field.

In fact we leverage this during syncing logic (to ensure we only sync when changes are made)

<details><summary>See logic matrix</summary>

Also documented in our [PRD](https://docs.google.com/document/d/1nXDYQWr5xnHxq_FR2ZYBkPhNx0XhCclFrJ_H0A1l2ZE/edit?tab=t.0#bookmark=id.yjlj7lwubgme)

![Screenshot 2024-11-27 at 20 59 40](https://github.com/user-attachments/assets/ad9d1be6-b678-43f8-911d-34439bee14c0)

</details>

Reply:
Alright, sounds good!